### PR TITLE
fix: argocd-image-updater@v0.12.2 to fix #694

### LINKED ISSUE #694: `Error fetching metadata` when using `ghcr.io` for image subscription
Stage spec:
```
spec:
  promotionMechanisms:
    argoCDAppUpdates:
      - appName: morey-tech-demo-app-dev
        appNamespace: argocd
    gitRepoUpdates:
      - kustomize:
          images:
            - image: ghcr.io/morey-tech/demo-app
              path: env/dev
        readBranch: ''
        repoURL: 'https://github.com/morey-tech/demo-app-deploy'
        writeBranch: main
  subscriptions:
    repos:
      images:
        - repoURL: ghcr.io/morey-tech/demo-app
          updateStrategy: Latest
```

Error in status:
```
status:
  error: >-
    error syncing image repo subscriptions: error getting latest suitable tag
    for image "ghcr.io/morey-tech/demo-app": found no suitable version of image
    "ghcr.io/morey-tech/demo-app"
```

Error in controller logs:
```
time="2023-09-08T20:10:17Z" level=error msg="Error fetching metadata for morey-tech/demo-app:d5f4dd376524d10a7e0ca4b2b52c6426399c7749 - neither V1 or V2 or OCI manifest returned by registry: manifest unknown: OCI index found, but Accept header does not support OCI indexes"
```

Tested the same image pushed to Dockerhub and it works.
```
spec:
  subscriptions:
    repos:
      images:
      - repoURL: docker.io/moreytech/demo-app
        updateStrategy: Latest
status:
  availableFreight:
    - firstSeen: '2023-09-08T20:56:20Z'
      id: 315ebae132f84ff09d8e030aa4ed8ac6ef1cc145
      images:
        - repoURL: docker.io/moreytech/demo-app
          tag: 2bc20dfbfd6df6be278b9fd7bc187832839486ec
```

### PR DESCRIPTION
**Pull request by morey-tech**
Fixes #694

### CODE REVIEW

**Comment by morey-tech** [comment]
@krancour I don't have a local env. Could you give this a test? The config shown in the issue should still be valid for testing.

**Comment by krancour** [comment]
@morey-tech on it.

**Comment by krancour** [comment]
Still appears to be an issue:

```
error syncing image repo subscriptions: error getting latest suitable tag for image \"ghcr.io/morey-tech/demo-app\": found no suitable version of image \"ghcr.io/morey-tech/demo-app\""
```

**Comment by krancour** [comment]
Wait... you have force-pushed since I grabbed your branch. Let me try again.

**Comment by morey-tech** [comment]
The force push was just to pass DCO. Maybe I didn't do the go mod update correctly? Was does the controller log say?

**Comment by krancour** [comment]
```
time="2023-09-11T14:59:18Z" level=error msg="error fetching metadata for morey-tech/demo-app:d5f4dd376524d10a7e0ca4b2b52c6426399c7749: could not fetch metadata for manifest sha256:feed5706bb16182d68c31e65b6b83af20e5e4b2eb28c92fb534ad9e15d7fe8c0: parsing time \"\" as \"2006-01-02T15:04:05.999999999Z07:00\": cannot parse \"\" as \"2006\""
time="2023-09-11T14:59:18Z" level=error msg="error fetching metadata for morey-tech/demo-app:2bc20dfbfd6df6be278b9fd7bc187832839486ec: could not fetch metadata for manifest sha256:640bab9110249b2ec5c54f1105b688ae448fd2c01452941b572b529d9942cb4b: parsing time \"\" as \"2006-01-02T15:04:05.999999999Z07:00\": cannot parse \"\" as \"2006\""
time="2023-09-11T14:59:18Z" level=error msg="error fetching metadata for morey-tech/demo-app:latest: could not fetch metadata for manifest sha256:640bab9110249b2ec5c54f1105b688ae448fd2c01452941b572b529d9942cb4b: parsing time \"\" as \"2006-01-02T15:04:05.999999999Z07:00\": cannot parse \"\" as \"2006\""
time="2023-09-11T14:59:18Z" level=error msg="error syncing Stage: error syncing image repo subscriptions: error getting latest suitable tag for image \"ghcr.io/morey-tech/demo-app\": found no suitable version of image \"ghcr.io/morey-tech/demo-app\"" namespace=kargo-demo stage=test
```

**Comment by morey-tech** [comment]
Probably an issue parsing dates here: https://github.com/akuity/kargo/blob/e6a23e16199f97bea60329f4a4d2625a4ab155ba/internal/images/image_updater.go#L138

**Comment by krancour** [comment]
Go date parsing works by providing examples of the format you are parsing. `2006-01-02T15:04:05.999999999Z07:00` is probably that example.

**Comment by krancour** [comment]
I just whipped up a standalone program to test this, and oddly it's working with v0.12.2.

I will try and figure out what v0.12.2 isn't doing the trick here.

**Comment by krancour** [comment]
Figured it out. There are layers of issues here:

Your original issue in #694 _is_ solved by this PR, so I am going to merge it.

The _new_ issue parsing the date, as near as I can tell is a _misleading_ error message coming from somewhere within image updater.

_I say it's misleading because image updater genuinely did not find what you were looking for..._

You did not specify an os/arch, so it defaulted to the native os/arch. I'm pretty sure you're on an M1 Mac, so "native" is the os/arch of the Docker Desktop VM -- linux/arm64.

_Your image doesn't exist for arm64_, so no image was found and the only real issue is the misleading error message -- you might consider opening an issue about that upstream.

This gist reproduces the issue: https://gist.github.com/krancour/1d4ff01e513e7cbf8ed1f0727d13ee67

**Review by krancour — APPROVED** [review]
